packages/core/admin/admin/src/components/Layouts/HeaderLayout.tsx

```diff
@@ -124,13 +195,24 @@ const HeaderLayout = (props: HeaderLayoutProps) => {
     }
   }, [baseHeaderLayoutRef]);
 
+  React.useEffect(() => {
+    const fetchDocLink = async () => {
+      const result = await getMatchingDocLink(pathname);
+      setDocLink(result);
+    };
+
+    fetchDocLink();
+  }, [pathname]);
```

Comment:
Since getMatchingDocLink doesn't need to be async, you can get the docLink by deriving it from the pathname directly in the component, no need for an effect or any state:

```
const { pathname } = useLocation();
const docLink = getMatchingDocLink(pathname);
```